pkg/plugin/processor/builtin/avro/encode.go

```diff
@@ -0,0 +1,165 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:generate paramgen -output=encode_paramgen.go encodeConfig
+//go:generate mockgen -source encode.go -destination=mock_encoder.go -package=avro -mock_names=encoder=MockEncoder . encoder
+
+package avro
+
+import (
+	"context"
+
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-processor-sdk"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/conduitio/conduit/pkg/foundation/log"
+	"github.com/conduitio/conduit/pkg/plugin/processor/builtin/avro/schemaregistry"
+	"github.com/goccy/go-json"
+	"github.com/lovromazgon/franz-go/pkg/sr"
+)
+
+type encoder interface {
+	Encode(ctx context.Context, sd opencdc.StructuredData) (opencdc.RawData, error)
+}
+
+type encodeProcessor struct {
+	sdk.UnimplementedProcessor
+
+	logger  log.CtxLogger
+	cfg     encodeConfig
+	encoder encoder
+}
+
+func NewEncodeProcessor(logger log.CtxLogger) sdk.Processor {
+	return &encodeProcessor{logger: logger}
+}
+
+func (p *encodeProcessor) Specification() (sdk.Specification, error) {
+	return sdk.Specification{
+		Name:    "encode.avro",
```

Comment:
In the design doc we said we should have `encode.avro` and `decode.avro`, this would mean that the code would need to live in the package `encode`. Obviously, it makes more sense to have both processors in the same package as they are sharing functionality, so I'm thinking if we should rather rename the processor to `avro.encode` and `avro.decode`? Same for the JSON processor. CC @maha-hajja

Reply:
Makes sense to me. :+1: